packages/code-block/partials/code-lines/index.js

```diff
@@ -59,7 +59,8 @@ function CodeLines({
                   isNotHighlighted={isNotHighlighted}
                   hasFloatingCopyButton={hasFloatingCopyButton}
                 >
-                  {lineChildren == '' ? '\n' : lineChildren}
+                  {lineChildren}
+                  {'\n'}
```

Comment:
Dropping a newline at the end of each line does not affect how the code is displayed (since there is no character following the newline), but does ensure that newline characters will be copied when the code is selected.

Thankfully this seems to target the Firefox issue well - in Safari and Chrome, which were already working, this does NOT seem to introduce extra newline characters 👍

Reply:
Sneaky 👀  good fix